pkg/containertools/command.go

```diff
@@ -0,0 +1,133 @@
+package containertools
+
+import (
+	"fmt"
+	"os/exec"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// Podman cli tool
+	Podman = "podman"
+	// Docker cli tool
+	Docker = "docker"
+)
+
+// CommandRunner defines methods to shell out to common container tools
+type CommandRunner interface {
+	GetToolName() string
+	Pull(image string) error
+	Build(dockerfile, tag string) error
+	Save(image, tarFile string) error
+	Inspect(image string) ([]byte, error)
+}
+
+// CommandRunnerImpl is configured to select a container cli tool and execute commands with that
+// tooling.
+type CommandRunnerImpl struct{
```

Comment:
I kind of expected a `DockerCommandRunner` and a `PodmanCommandRunner` implementation - I know their cli's are supposed to be compatible, but are they the only two we may support? Are there no differences other than the root command?

(also not blocking; more a question for future changes)

Reply:
That was my first thought too, but it turned out that so far the implementation is identical so it seemed like a bit of a waste to create two implementations of the interface. I was imagining that if/when the actual implementation differed we could split it out at that point.